easybuild/easyblocks/o/openmpi.py

```diff
@@ -164,4 +175,33 @@ def sanity_check_step(self):
 
         custom_commands = ["%s --version | grep '%s'" % (key, expected[key]) for key in sorted(expected.keys())]
 
+        # Add minimal test program to sanity checks
+        # Run with correct MPI launcher
+        mpi_cmd_tmpl, params = get_mpi_cmd_template(toolchain.OPENMPI, dict(), mpi_version=self.version)
+        for src, compiler in (('hello_c.c', 'mpicc'), ('hello_mpifh.f', 'mpifort'), ('hello_usempi.f90', 'mpif90')):
+            src_path = os.path.join(self.cfg['start_dir'], 'examples', src)
+            if os.path.exists(src_path):
+                test_exe = os.path.join(self.builddir, 'mpi_test_' + os.path.splitext(src)[0])
+                self.log.info("Adding minimal MPI test program to sanity checks: %s", test_exe)
+
+                # Build test binary
+                build_cmd = "%s %s -o %s" % (compiler, src_path, test_exe)
+
+                # Skip running the test if chosen
+                if not build_option('mpi_tests'):
+                    continue
```

Comment:
Shouldn't this rather be above the for-loop?

Reply:
Good catch! It is actually only supposed to skip running the tests, not compiling them. Changed that in https://github.com/easybuilders/easybuild-easyblocks/pull/2444/commits/3bc992371f74554d382b4fba13b5cc1e5f8c049f